FunctionalTableData/TableSectionHeaderFooter.swift

```diff
@@ -15,9 +15,16 @@ public typealias TableHeaderConfigType = TableHeaderFooterConfigType
 
 public protocol TableHeaderFooterConfigType: TableItemConfigType {
 	func dequeueHeaderFooter(from tableView: UITableView) -> UITableViewHeaderFooterView?
+	func isEqual(_ other: TableHeaderFooterConfigType?) -> Bool
 	var height: CGFloat { get }
 }
 
+public extension TableHeaderFooterConfigType {
+	func isEqual(_ other: TableHeaderFooterConfigType?) -> Bool {
+		return (other as? Self) != nil
```

Comment:
Why does this have to be cast to `Self` if the parameter is already constrained to an identical type? Can this just be:

```suggestion
		return other != nil
```

Reply:
No, if we had the following:
```
struct Header1: TableSectionHeaderConfigType {}
struct Header2: TableSectionHeaderConfigType {}
```

 at runtime the type of `Self` would evaluate to one of `Header1` or `Header2`, but the type of `other` evaluates to `TableSectionHeaderConfigType?`. We need to cast `other` to `Self` to ensure that we have two instances of `Header1`, and not a instance of `Header1` and an instance of `Header2`.